ops/charm.py

```diff
@@ -1376,6 +1379,12 @@ def __init__(self, framework: Framework):
                 f'{container_name}_pebble_check_recovered', PebbleCheckRecoveredEvent
             )
 
+    def _destroy_charm(self):
+        for event in self.on.events():
+            if event in self._static_events:
+                continue
```

Comment:
`set(...) - set(...)` may be more expressive.

Reply:
I'm not sure. It's briefer (one set of parentheses might not be required here):

```suggestion
        for event in (self._static_events - set(self.on.events())):
```

Personally, I feel like having the if/continue is more readable, as well as avoiding creating an additional set (although set-set does avoid the O(n) checking for membership).